src/main/java/Client/ConsoleClientLauncher.java

```diff
@@ -28,17 +27,25 @@
     private static final Logger logger = LoggerFactory.getLogger(ConsoleClientLauncher.class);
     public static Future<Void> clientListening;
 
+    static {
+        socket = null;
+        client = null;
+        menu = null;
+    }
+
     public ConsoleClientLauncher(String host, int port) throws IOException {
         ConsoleClientLauncher.host = host;
         ConsoleClientLauncher.port = port;
         client = new MopeLSPClient();
         socket = new Socket(host, port);
         menu = new ConsoleMenu(client);
     }
+    public static void setExecutorPool() {
+        executor = Executors.newFixedThreadPool(2);
+    }
```

Comment:
yes, it also was referring of a static field (executor) in a non static context, but i guess you already find out because of the further feedback below. Lots of work to do for me ;-)